js/app/controllers/generic-csv-importer.js

```diff
@@ -116,62 +156,70 @@
-										// console.error(e);
+										console.error(e);
+										continue;
 									}
-								} else {
-									for(let j = 0; j < row[k].length; j++){
-										if (row[k][j].field_type === 'file'){
-											var _file = {
-												filename: row[k][j].value.filename,
-												size: row[k][j].value.size,
-												mimetype: row[k][j].value.mimetype,
-												data: row[k][j].value.file_data
-											};
-
-											row[k][j].value = await FileService.uploadFile(_file).then(FileService.getEmptyFileWithDecryptedFilename);
+								}
+								for(let j = 0; j < row[k].length; j++){
+									if (row[k][j].field_type === 'file'){
+										const _file = {
+											filename: row[k][j].value.filename,
+											size: row[k][j].value.size,
+											mimetype: row[k][j].value.mimetype,
+											data: row[k][j].value.file_data ? row[k][j].value.file_data : row[k][j].value.data
+										};
+										if (_file.data === undefined) {
+											console.error('Unable to parse file data from ', row[k][j]);
+											$scope.log.push('Unable to parse file data from file ' + _file.filename);
+											continue;
 										}
-										_credential.custom_fields.push(row[k][j]);
+										row[k][j].value = await FileService.uploadFile(_file).then(FileService.getEmptyFileWithDecryptedFilename);
 									}
+									_credential.custom_fields.push(row[k][j]);
 								}
 							} else if(field === 'files'){
 								if (row[k] !== undefined && (typeof row[k] === 'string' || row[k] instanceof String) && row[k].length > 1){
 									try {
 										row[k] = JSON.parse(row[k]);
-										for(let i = 0; k < row[k].length; i++){
-											_credential.files.push({
+										for(let i = 0; i < row[k].length; i++){
+											_credential.files.push(await FileService.uploadFile({
 												filename: row[k][i].filename,
 												size: row[k][i].size,
-												mimetype: row[k][i].mimetype
-											});
+												mimetype: row[k][i].mimetype,
+												data: row[k][i].file_data ? row[k][i].file_data : row[k][i].data
+											}).then(FileService.getEmptyFileWithDecryptedFilename));
 										}
 									} catch (e) {
 										// ignore row[k], it contains no valid json data
-										// console.error(e);
+										console.error(e);
 									}
 								} else {
 									for(let j = 0; j < row[k].length; j++){
 										_credential.files.push(await FileService.uploadFile({
 											filename: row[k][j].filename,
 											size: row[k][j].size,
 											mimetype: row[k][j].mimetype,
-											data: row[k][j].file_data
+											data: row[k][j].file_data ? row[k][j].file_data : row[k][j].data
 										}).then(FileService.getEmptyFileWithDecryptedFilename));
 									}
 								}
 							} else if(field === 'tags'){
-								if( row[k]) {
-									var tags = row[k].split(',');
+								if(row[k] && row[k] !== '' && row[k] !== '[]') {
```

Comment:
Why not just JSON.Parse(row[k]) instead of all this logic?

Reply:
because a string like this `"[mytg1,mtg2]"` (produced by the current exporter) is not json parsable and produces a SyntaxError

Reply:
ah, makes sense :)